### ISSUE TITLE
Middle labels on x-axis aren't rendered when rotation is set to 180

### ISSUE THREAD

**Issue by jedrzejruta** [highcharts/highcharts#19240]
#### Expected behaviour
Labels are rendered and evenly spaced out when there is enough room for them.

#### Actual behaviour
Only first and last label is rendered, even with a small font size set.

#### Live demo with steps to reproduce
[https://jsfiddle.net/BlackLabel/Lns7694e/](https://jsfiddle.net/BlackLabel/Lns7694e/)

#### Product version
Highcharts 11 and pressumably previous major versions, checked up to Highcharts 8.

#### Affected browser(s)
All, this issue doesn't seem to be browser related.

**Comment by karolkolodziej** [comment]
As a workaround please specify the ticks- [demo](https://jsfiddle.net/BlackLabel/kz1rw539/).

**Comment by vitaliy-kretsul** [comment]
Hi @karolkolodziej 
Thanks for the workaround. It might be helpful in some cases, but doesn't help in cases when you have long categories names, because when 'tickPositions' is set, the logics for skipping intermediate xaxis ticks doesn't seem to work:
https://jsfiddle.net/Cortez/jfk5zh2y/

Also, want to add that this problem is relevant not only for rotation 180. For me this problem is happening somewhere between ~167 and ~194. Not all xaxis ticks are visible, despite all of them can perfectly fit since there is plenty of room.

**Comment by karolkolodziej** [comment]
Yes, you are right @vitaliy-kretsul! This needs further investigation.

This also might be a clue for the future that if the rotation is set the ticks are missing.
